### PR TITLE
Update to 20.08 version of the ethosn-driver.

### PR DESCRIPTION
**Pull request by tristan-arm**
- Updated ethosn relay backend to account for 20.08 api changes around cascading and quantization.
   - Note: 20.05 compatibility is maintained for now to avoid compilation and test failures while the docker image still uses 20.05.

### CODE REVIEW

**Review by mbaret — APPROVED** [review]
This lgtm, but I think we should add a bit of context on why we need to do this and what our plan is moving forward so that other's can reason about whether this is best practice. Could you add this to the commit message + as a comment in this review?

**Comment by mbaret** [comment]
cc @Leo-arm @u99127 @comaniac @tqchen

File: tests/python/contrib/test_ethosn/test_conv2d.py [hunk 1]
```diff
@@ -185,6 +185,14 @@ def test_conv2d_failure():
     if not ethosn_available():
         return
 
+    _scale_error_msg = (
+        "Overall scale (of the input * weights / output) should be in the range [0, 1)"
+    )
+    if tei.get_ethosn_api_version() == 2008:
+        _scale_error_msg = (
+            "Overall scale (of the input * weights / output) should be in the range [0, 1}"
+        )
```
Comment (comaniac) [inline tests/python/contrib/test_ethosn/test_conv2d.py:194]: Are you trying to match the error message from the Ethos-N compiler? If so, it seems to me that we should be fine as long as we could detect the error (by error code or any other form).

If you really need to check the error message for whatever reasons, we shouldn't be bothered by the small message changes. For example, you can just match `msg.startswith("Overall scale")`.

**Review by comaniac — APPROVED** [review]
1. Please note that CI passing for this PR means the changes are compatible with 20.05 but not 20.08. In case there are some issues with 20.08, we won't find it until we update the docker image in the future.

2. Most changes highly depend on the tool, so I just reviewed the test cases. We really need to figure out a better way to improve this situation.

Also cc @zhiics

**Review by u99127 — APPROVED** [review]
LGTM , modulo Cody's testsuite comments. 

Presumably @tristan-arm you've locally tested this and it works with a 20.08 installation under the hood as far as just now. 

My expectation is that over time we'd drop the older versions support and continually support the latest and greatest version but we need something like this to tide us through until all work can switch over to the latest and greatest release. .

**Comment by u99127** [comment]
Oh and @tqchen , could we request a update the frozen Docker CI image shortly so that we don't have too much impedance mismatch between this and the rest of the stack ?

**Comment by comaniac** [comment]
Thanks @tristan-arm @mbaret @u99127.

**Pull request merged**